Springer Link.js

```diff
@@ -31,185 +29,191 @@ function detectWeb(doc, url) {
 			} else {
 				return false;
 			}
-		break;
+			break;
 		case "article":
 			return "journalArticle";
-		break;
+			break;
 		case "chapter":
 		case "referenceworkentry":
 		case "protocol":
 			return "bookSection";
-		break;
+			break;
 	}
 }
 
 function getResultList(doc) {
-	var results =  ZU.xpath(doc,
-		'//ol[@class="content-item-list"]/li/*[self::h3 or self::h2]/a');
+	var results = ZU.xpath(doc, '//ol[@class="content-item-list"]/li/*[self::h3 or self::h2]/a');
 	if(!results.length) {
-		results = ZU.xpath(doc,
-			'//div[@class="toc"]/ol//div[contains(@class,"toc-item")]/h3/a');
+		results = ZU.xpath(doc, '//div[@class="toc"]/ol//div[contains(@class,"toc-item")]/h3/a');
 	}
 	if(!results.length) {
 		results = ZU.xpath(doc, '//div[@class="toc"]/ol\
 			//li[contains(@class,"toc-item")]/p[@class="title"]/a');
 	}
-
 	return results;
 }
 
-function doWeb(doc, url){ 
+function doWeb(doc, url) {
 	var type = detectWeb(doc, url);
 	if(type == "multiple") {
 		var list = getResultList(doc);
 		var items = {};
-		for(var i=0, n=list.length; i<n; i++) {
+		for(var i = 0, n = list.length; i < n; i++) {
 			items[list[i].href] = list[i].textContent;
 		}
-
 		Zotero.selectItems(items, function(selectedItems) {
 			if(!selectedItems) return true;
-
 			for(var i in selectedItems) {
 				ZU.processDocuments(i, scrape);
 			}
 		})
 	} else {
-		scrape(doc)
+		scrape(doc, url)
 	}
 }
 
-function scrape(doc) {
+function complementItem(doc, item) {
 	var itemType = detectWeb(doc, doc.location.href);
-	
-	//use Embeded Metadata translator
-	var translator = Zotero.loadTranslator("web");
-	translator.setTranslator("951c027d-74ac-47d4-a107-9c3069ab7b48");
-	translator.setDocument(doc);
-		
-	translator.setHandler("itemDone", function(obj, item) {
-		//sometimes we get an error about title not being set
-		if(!item.title) {
-			Z.debug("Springer Link: title not found");
-			Z.debug(item);
-			if(doc.head) {
-				//clean up and strip out uninteresting content
-				Z.debug(doc.head.innerHTML
-					.replace(/<style[^<]+(?:<\/style>|\/>)/ig, '')
-					.replace(/<link[^>]+>/ig, '')
-					.replace(/(?:\s*[\r\n]\s*)+/g, '\n'));
-			} else {
-				Z.debug("Springer Link: no head tag");
-			}
-		}
-		
-		//in case we're missing something, we can try supplementing it from page
-		if(!item.DOI) {
-			item.DOI = ZU.xpathText(doc,
-				'//dd[@id="abstract-about-book-chapter-doi"\
+	//in case we're missing something, we can try supplementing it from page
+	if(!item.DOI) {
+		item.DOI = ZU.xpathText(doc,
+			'//dd[@id="abstract-about-book-chapter-doi"\
 					or @id="abstract-about-doi"][1]');
+	}
+	if(!item.publisher) {
+		item.publisher = ZU.xpathText(doc, '//dd[@id="abstract-about-publisher"]');
+	}
+	if(!item.date) {
+		item.date = ZU.xpathText(doc, '//dd[@id="abstract-about-cover-date"]') || ZU.xpathText(doc,
+			'//dd[@id="abstract-about-book-chapter-copyright-year"]');
```

Comment:
I relied on jsbeautify (set to 110 chars max) entirely. I personally think that's good practice because that allows future versions to just be passed through the same beautifier w/o messing up the diff. (also applies to "long line" above). 
I'm OK adjusting the char length (I thought this worked well, though), but I'd rather not edit the automatically chosen linebreaks.